storage/s3.go

```diff
@@ -294,6 +294,30 @@ func (s *S3) listObjects(ctx context.Context, url *url.URL) <-chan *Object {
 	return objCh
 }
 
+// setAcl sets value of acl flag pointer according to the metadata
+// provided.
+func setAcl(acl **string, metadata map[string]string) error {
```

Comment:
2-level indirection doesn't look very readable to me.

While we're here, let's create a new type

`type Metadata map[string]string`

and define methods on that type.

`func (m Metadata) SetACL(acl string)`
`func (m Metadata) SetStorageClass(class string)`
`func (m Metadata) ACL() string`
`func (m Metadata) StorageClass() string`
etc.

would be more safe and readable.

Reply:
great, thanks! will do that.